saleor/payment/gateways/braintree/errors.py

```diff
@@ -117,3 +127,28 @@ def get_error_type(error_code):
 
 class BraintreeException(Exception):
     pass
+
+
+def handle_braintree_error(exc):
+    if isinstance(exc, AuthenticationError):
+        msg = "Authentication error"
+    elif isinstance(exc, AuthorizationError):
+        msg = "Authorization error"
+    elif isinstance(exc, NotFoundError):
+        msg = DEFAULT_ERROR_MESSAGE
+    elif isinstance(exc, RequestTimeoutError):
+        msg = "Request timeout error"
+    elif isinstance(exc, UpgradeRequiredError):
+        msg = "Upgrade required error"
+    elif isinstance(exc, TooManyRequestsError):
+        msg = "Too many requests error"
+    elif isinstance(exc, ServerError):
+        msg = "Server error"
+    elif isinstance(exc, ServiceUnavailableError):
+        msg = "Server unavailable error"
+    elif isinstance(exc, GatewayTimeoutError):
+        msg = "Gateway timeout error"
```

Comment:
IMO, the better and clearer solution is using mapping here, where keys are errors and values are messages. Then we can just check if `exc` is in the given mapping and return the correct msg if is, or default if isn't.